it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationControllerISpec.scala

```diff
@@ -544,6 +595,148 @@ trait BusinessIdentificationControllerISpec extends BaseISpec with SessionDataMi
+
+      val result = await(controller.submitBusinessNameForm(request))
+
+      result should containMessages("businessName.title", "error.business-name.empty")
+    }
+
+    "redirect to the /business-type page if there is no InitialDetails in session because the user has returned to a bookmark" in {
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments)
+
+      val result = await(controller.submitBusinessNameForm(request))
+
+      resultShouldBeSessionDataMissing(result)
+    }
+  }
+
+  "showBusinessEmailForm" should {
+    behave like anAgentAffinityGroupOnlyEndpoint(request => controller.showBusinessEmailForm(request))
+
+    "display business email form" in {
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments)
+      sessionStoreService.currentSession.initialDetails = Some(initialDetails)
+
+      val result = await(controller.showBusinessEmailForm(request))
+      result should containMessages("businessEmail.title", "businessEmail.description", "businessEmail.continue.button")
+      val doc = Jsoup.parse(bodyOf(result))
+      doc.getElementById("email").`val` shouldBe initialDetails.email.get
+
+      val form = doc.select("form").first()
+      form.attr("method") shouldBe "POST"
+      form.attr("action") shouldBe routes.BusinessIdentificationController.submitBusinessEmailForm().url
+    }
+
+    "redirect to the /business-type page if there is no InitialDetails in session because the user has returned to a bookmark" in {
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments)
+
+      val result = await(controller.showBusinessNameForm(request))
+
+      resultShouldBeSessionDataMissing(result)
+    }
+  }
+
+  "submitBusinessEmailForm" should {
+    behave like anAgentAffinityGroupOnlyEndpoint(request => controller.submitBusinessEmailForm(request))
+
+    "update business email after submission" in {
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments).withFormUrlEncodedBody(
+        "email" -> "newagent@example.com")
+      sessionStoreService.currentSession.initialDetails = Some(initialDetails)
+
+      val result = await(controller.submitBusinessEmailForm(request))
+      status(result) shouldBe 303
+      redirectLocation(result).head shouldBe routes.SubscriptionController.showCheckAnswers().url
+
+      await(sessionStoreService.fetchInitialDetails).get.email shouldBe Some("newagent@example.com")
+    }
+
+    "show validation error when the form is submitted with empty email" in {
+      implicit val request =
+        authenticatedAs(subscribingCleanAgentWithoutEnrolments).withFormUrlEncodedBody("email" -> "")
+      sessionStoreService.currentSession.initialDetails = Some(initialDetails)
+
+      val result = await(controller.submitBusinessEmailForm(request))
+
+      result should containMessages("businessEmail.title", "error.business-email.empty")
```

Comment:
Another version of this test, but for the pre-check-answers version of the page?
e.g. with an assertion like `result should containMessages("businessEmail.updated.title")`